csrc/opaque_type.h

```diff
@@ -79,7 +82,11 @@ class Opaque {
 
 template <typename T>
 bool OpaqueEquals<T>::operator()(const Opaque& a, const Opaque& b) const {
-  return a.as<T>() == b.as<T>();
+  if constexpr (dynamic_type::opcheck<T> == dynamic_type::opcheck<T>) {
```

Comment:
Yes. It returns false at least for the test case. Just feel the format is confusing. Thanks for the detailed explain. Can we add what you wrote as comment to the code:

> opcheck is specifically designed for the existence of operators, so the operators on opcheck behave differently from normal C++. For example, if you assign one opcheck to another opcheck, it will return a bool telling you whether this asignment is valid for the underlying type, instead of returning a reference of the lhs opcheck.

There are some comments on `opcheck` but seems out of date. e.g. `// For more examples, see test_dynamic_type.cpp namespace opcheck_tests`

Reply:
comment added